org.scala-ide.sdt.core/src/org/scalaide/core/internal/jdt/model/ScalaElements.scala

```diff
@@ -81,42 +80,44 @@ class ScalaSourceTypeElement(parent: JavaElement, name: String, declaringType: O
   }
 
   override def getFullyQualifiedName: String =
-    declaringType.map { declaringType =>
-      val pkgSym = declaringType.typeSymbol.enclosingPackage
-      if (pkgSym.isEmptyPackage)
-        super.getFullyQualifiedName
-      else {
-        val pkg = pkgSym.javaClassName
-        pkg + "." + getTypeQualifiedName('$', /*showParameters =*/ false)
-      }
+    declaringType.flatMap { typ =>
+      pc.asyncExec {
+        val pkgSym = typ.typeSymbol.enclosingPackage
+        if (pkgSym.isEmptyPackage)
+          super.getFullyQualifiedName
+        else {
+          val pkg = pkgSym.javaClassName
+          pkg + "." + getTypeQualifiedName('$', /*showParameters =*/ false)
+        }
+      }.get.left.toOption
```

Comment:
If you do a `import org.scalaide.core.compiler.IScalaPresentationCompiler.Implicits._` you can do `.getOption()`, which has the advantage that a timeout will be used.